### ISSUE TITLE
Fix S4502 FN: Support generic attributes

### ISSUE THREAD

**Issue by csaba-sagi-sonarsource** [SonarSource/sonar-dotnet#6314]
### Description

FP for [S4502](https://sonarsource.github.io/rspec/#/rspec/S4502/csharp) (DisablingCsrfProtection).

Using generic attributes that inherit from IgnoreAntiforgeryTokenAttribute the rule is not  raising.

### Repro steps

```
    internal class TestCases
    {
        [GenericIgnoreAntiforgeryToken<int>] // FN
        public void A() { }

        public void C() { }
    }
    public class GenericIgnoreAntiforgeryToken<T> : IgnoreAntiforgeryTokenAttribute { }
```

### Expected behavior

The rule should raise

### Actual behavior

The rule is not raising.

**Comment by gregory-paidis-sonarsource** [comment]
We decided to not support generic attributes, as well as class inheritance in general, since the benefits/frequency of usecases do not outweight the performance hit.
